lib/puppet/parser/functions/new.rb

```diff
@@ -864,7 +866,7 @@
 | b | The data is in base64 encoding, padding as required by base64 strict is added by default
 | u | The data is in URL safe base64 encoding
 | B | The data is in base64 strict encoding
-| s | The data is a puppet string used verbatim as a binary byte sequence
+| s | The data is a puppet string seen as a binary sequence of bytes with ignored encoding
```

Comment:
Do you agree that '%s' should use the same encoding when converting to/from a Binary? If the answer to that is yes, then "%s" has to mean "UTF-8". Some other format must then be added to mean "whatever" or, perhaps "ISO-8859-1". As you mentioned before, it's a valid encoding that will accept the full 8-bit range of bytes. I wouldn't be surprised if it's a no-op in Ruby (i.e. bytes == string characters).